precedence.js

```diff
@@ -0,0 +1,53 @@
+(function() {
+    "use strict";
+
+    function Precedence() {
+        this.numbers = [];
+        this.operations = [];
+    }
+
+    Precedence.prototype.calculateResult = function() {
+        var temporaryResult = this.numbers[0];
+        var finalResult = 0;
+        for (var i = 0; i < this.operations.length; i++) {
+            if (this.operations[i] === 'x') {
+                temporaryResult *= this.numbers[i + 1];
+            } else if (this.operations[i] === '/') {
+                temporaryResult /= this.numbers[i + 1];
+            } else if (this.operations[i] === '-') {
```

Comment:
Logic for both '-' and '+' is pretty similar, almost copy / paste, except only sign differs. Could we do everything in a more elegant way. Here is a hint:
```
} else if (i + 1 < this.operations.length) {
    var sign = this.operations[i] === '+' ? 1 : -1;
    if (this.operations[i + 1] === '/' || this.operations[i + 1] === 'x') {
        finalResult += temporaryResult;
        temporaryResult += sing * this.numbers[i + 1];
    } else { ... }
}
```

Reply:
Btw nice effort, now it looks waaay sexier.